plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/database/DatabaseMappingAttribute.java

```diff
@@ -338,10 +338,10 @@ private void changeDataTypeLength(@NotNull DBPDataSource targetDataSource) {
                     preferenceStore.getInt(DTConstants.PREF_MAX_TYPE_LENGTH) : store.getInt(DTConstants.PREF_MAX_TYPE_LENGTH);
                 DBSTypedObjectExt2 sourceExt = (DBSTypedObjectExt2) source;
                 if (source.getDataKind() == DBPDataKind.NUMERIC &&
-                    (source.getPrecision() == null || source.getPrecision() > maxDataTypeLength)
+                    source.getPrecision() != null && source.getPrecision() > maxDataTypeLength
```

Comment:
Logic is changed here. Before that unspecified precision triggered max length set

Reply:
Yes, because we do not want to set precision for values without it (for the smallint() without modifiers, as example)